Switch to  gicv3

dsemenets opened the pull request.

oleksiimoisieiev commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -28,6 +28,7 @@
 		zephyr,sram = &ram;
 		zephyr,console = &xen_consoleio_hvc;
```
Why Flash SIMULATOR is in this MR?

oleksiimoisieiev commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -82,4 +83,45 @@
 		status = "okay";
 	};
+	flashcontroller0: flashcontroller {
+		compatible = "zephyr,sim-flash";
```
I think we still need to be able to select which gic do we need.

oleksiimoisieiev commented on boards/arm64/xenvm/xenvm_defconfig:
```diff
@@ -33,9 +34,23 @@ CONFIG_MAIN_STACK_SIZE=32768
+#CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
+#CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
+#CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
+#CONFIG_IPM_CONSOLE_STACK_SIZE=32768
```
Please do not push commented code. It's not obvious what are these changes for.

oleksiimoisieiev commented on soc/arm64/xenvm/soc.h:
```diff
@@ -0,0 +1,4 @@
+#ifndef _SOC_H
+#define _SOC_H
```
Why do we need this empty header?

oleksiimoisieiev commented:
Also please update commit messages. Make them complete, so your commit can be cherry-picked to features.

oleksiimoisieiev commented on boards/arm64/xenvm/board.cmake:
```diff
@@ -3,13 +3,14 @@ set(QEMU_ARCH aarch64)
 set(QEMU_FLAGS_${ARCH}
    -gdb tcp::4224
    -cpu ${QEMU_CPU_TYPE_${ARCH}} -m 6G
    -nographic
```
I think we still need to make GIC verision optional. I see this implementation as to create separate configs to handle both gicv2 and gicv3

oleksiimoisieiev commented:
I think there should be some parameter, exposed to the code so correct gic version can be placed in the domain_config.

oleksiimoisieiev commented:
I suggest renaming CONFIG_XENVM_USE_GIC_V3 to CONFIG_XEN_USE_GIC_V3 so this config can be used for the future configs. Also I think CONFIG_XEN_USE_GIC_V3 should be mentioned in KConfig with appropriate dependencies.

dsemenets commented:
> I suggest renaming CONFIG_XENVM_USE_GIC_V3 to CONFIG_XEN_USE_GIC_V3
I don't think it is reasonable. This option created for xenvm board not for xen subsystem

firscity commented on boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:
```diff
@@ -0,0 +1,31 @@
+CONFIG_SOC_XENVM=y
```
Do we actually need whole new defconfig for just 1 changed option? I think it is better to left default GICv3 for xenvm_qemu and, if someone will need GICv2 in qemu, just change this value on own setup.

dsemenets replied:
There is more then 1 option.
There is different reg value in dts. I wanted to do this switcher by option but I can't use different dts for this. Some reason dts processed before any of CMakeList.txt I can change (in boards/arm64/xenvm/ and application directories)

firscity reviewed (commented):
Please, no short or empty commit messages. Such commits will not be merged.

firscity commented on boards/arm64/xenvm/board.cmake:
```diff
@@ -2,14 +2,18 @@ set(SUPPORTED_EMU_PLATFORMS qemu)
    -cpu ${QEMU_CPU_TYPE_${ARCH}} -m 6G
    -nographic
-   -machine virt,gic-version=2,virtualization=true
+   -machine virt,gic-version=${QEMU_GIC_VERSION},virtualization=true
```
Are this changes needed if we decided to build/run qemu separately from Zephyr (build with Moulin/run with separate script)? I don`t think that Zephyr OS kernel is proper place for this. Also this will be broken with zephyr-dom0-xt, because it doesn't build Xen.

dsemenets replied:
I did this before we decided that. Other hand I had to make test. As far as I know we didn't move to run qemu separately from Zephyr

firscity commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -11,75 +11,9 @@
  * with zephyr requirements.
```
This description is not correct anymore. Also copyright should be `2023`, it was not created in 2020

firscity commented on boards/arm64/xenvm/xenvm-gicv2-qemu.dts:
```diff
@@ -0,0 +1,19 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
This is not true, please, remove this comment and set copyright to 2023.

firscity commented on boards/arm64/xenvm/xenvm-qemu.dtsi:
```diff
@@ -0,0 +1,86 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
Copyright and comment again.

firscity commented on boards/arm64/xenvm/xenvm-qemu.dtsi:
```diff
@@ -0,0 +1,86 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This file was created by running
```
This file was NOT created by LIBXL_DUMP_DTB, it is manually created for qemu Dom0. Please, remove or edit this comment.

dsemenets replied:
Done

firscity commented on boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:
```diff
@@ -0,0 +1,31 @@
+CONFIG_UART_CONSOLE=y
+
+# Enable PV hypervisor console
+#CONFIG_UART_CONSOLEIO_XEN_HVC=y
```
Please fix this, no commented configs, this issue is already fixed in this branch. Please, check the correct name for this config.

dsemenets replied:
Done

firscity commented on boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:
```diff
@@ -0,0 +1,31 @@
+CONFIG_LOG_MODE_MINIMAL=n
+
+CONFIG_TEST=y
+#CONFIG_PRINTK_SYNC=y
```
No commented configs.

dsemenets replied:
Done

firscity reviewed (changes requested):
Also, no empty or short commit messages are allowed.

dsemenets committed:
xenvm-qemu: run qemu with proper gic version

This is trmporary solution and can be removed when build will be
performed with moulin

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

firscity commented on boards/arm64/xenvm/xenvm-gicv2-qemu_defconfig:
```diff
@@ -0,0 +1,30 @@
+CONFIG_UART_CONSOLE=y
+
+# Enable PV hypervisor console
+CONFIG_UART_CONSOLEIO_XEN_HVC=y
```
This code is not buildable.` CONFIG_UART_CONSOLEIO_XEN_HVC `-> `CONFIG_UART_XEN_HVC_CONSOLEIO`

firscity commented on boards/arm64/xenvm/xenvm-qemu_defconfig:
```diff
@@ -26,7 +26,6 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_TEST=y
-#CONFIG_PRINTK_SYNC=y
```
Looks like spurious change. It is right, but at least it should be in separate commit.

firscity commented:
@dsemenets, please add more appropriate commit message for commit 1 and 2. It should contain about motivation and point of the changes, not about copying commit topic (at least 1 paragraph of text, as was described by @lorc some time ago).

Commit 3 will be dropped from git history, so it may be unchanged.

dsemenets committed:
xenvm-qemu: add xenvm-gicv3-qemu board

This patch introduces xenvm-giv3-qemu board. The only difference with
xenvm-qemu new board configured for use gic-v3 instead gic-v2

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

firscity commented on boards/arm64/xenvm/xenvm-gicv3-qemu_defconfig:
```diff
@@ -0,0 +1,34 @@
+CONFIG_LOG_MODE_MINIMAL=n
+
+CONFIG_TEST=y
+#CONFIG_PRINTK_SYNC=y
```
Please remove commented config

firscity commented on boards/arm64/xenvm/xenvm-qemu.dts:
```diff
@@ -1,85 +1,13 @@
-		interrupt-parent = <&gic>;
-	};
+#include "xenvm-qemu.dtsi"
 
```
Redundant whitespace

oleksiimoisieiev commented:
Why did you add gicv2/3 switch to xenvm-qemu only? We need this to build xenvm as well. Also I think this switch can be done as overlay for both boards.

dsemenets commented:
We already have xenvm_gicv3 board for gicv3

firscity reviewed (approved):
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`
